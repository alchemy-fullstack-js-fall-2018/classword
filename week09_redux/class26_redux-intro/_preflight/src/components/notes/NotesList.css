@import '../../styles/variables.css';

.notes {
  h3 {
    margin: 0;
  }
  .notesList {
    display: flex;
    justify-content: start;
    flex-wrap: wrap;
    list-style: none;
    margin: 0;
    padding: 1em;
  }

  .noteItem {
    text-decoration: none;
    padding: 0.5em;
    border: 1px solid black;
    margin: 0.4em;

    &:visited {
      color: #000;
    }

    &:hover {
      background-color: var(--lightWhite);
    }

    &:active {
      color: #000;
    }
  }
}
