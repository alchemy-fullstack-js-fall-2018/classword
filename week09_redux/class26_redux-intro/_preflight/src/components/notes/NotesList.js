import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';
import styles from './Notes.css';

const Notes = ({ notes }) => {
  const notesComponents = notes.map(({ id, title }) =>
    <Link className={styles.noteItem} key={id} to={ROUTES.NOTE.linkTo(id)}>{title}</Link>);
  return (
    <div className={styles.notes}>
      <h3>Notes</h3>
      <div className={styles.notesList}>
        {notesComponents.length > 0 ?
          notesComponents :
          <p>no notes!</p>}
      </div>
    </div>
  );
};

Notes.propTypes = {
  notes: PropTypes.array.isRequired
};

export default Notes;
