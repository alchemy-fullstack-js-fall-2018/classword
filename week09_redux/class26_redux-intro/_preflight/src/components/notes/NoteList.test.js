import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import NotesList from './NotesList';

jest.mock('../../routes/routes.js', () => ({
  ROUTES: {
    NOTE: {
      linkTo: () => '/'
    }
  }
}));

describe('NotesList', () => {
  const notes = [{ id: '1234', title: 'hi there' }];

  it('renders a NoteList component', () => {
    const wrapper = shallow(<NotesList notes={notes} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('displays no notes if the notes list is empty', () => {
    const wrapper = shallow(
      <MemoryRouter>
        <NotesList notes={[]} />
      </MemoryRouter>);

    expect(wrapper.html()).toContain('no notes!');
  });

  it('displays notes if the notes is not empty', () => {
    const wrapper = shallow(
      <MemoryRouter>
        <NotesList notes={notes} />
      </MemoryRouter>);
    expect(wrapper.html()).toContain('hi there');
  });
});
