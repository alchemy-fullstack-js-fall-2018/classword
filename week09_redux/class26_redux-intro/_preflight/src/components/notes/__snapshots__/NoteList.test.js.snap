// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesList renders a NoteList component 1`] = `
<div
  className="notes"
>
  <h3>
    Notes
  </h3>
  <div
    className="notesList"
  >
    <Link
      className="noteItem"
      key="1234"
      replace={false}
      to="/"
    >
      hi there
    </Link>
  </div>
</div>
`;
